x/campaign/types/total_supply.go

```diff
@@ -21,3 +25,34 @@ func UpdateTotalSupply(coins, updatedCoins sdk.Coins) sdk.Coins {
 
 	return append(updatedCoins, notUpdated...).Sort()
 }
+
+// ValidateTotalSupply checks whether the total supply for each denom is within the provided total supply range
+func ValidateTotalSupply(coins sdk.Coins, supplyRange TotalSupplyRange) error {
+	// extra safety check, this should never happen
```

Comment:
I think it's actually a good practive to include these checks
In this case since it is an inconsistent state error, this should be handled by a `CriticalError`

Since the package `types` is stateless and unopinionated from `CriticalError`, this method should return a specific error type at this stage `ErrInvalidSupplyrange` then the calling methods in `keeper` should return themselves a `CriticalError` if `ErrInvalidSupplyrange` occurs